py-pytest-mpl: update to version 0.16.1 and add Python 3.10-3.11

nilason opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->
py-pytest-mpl: update to version 0.16.1 and add Python 3.10, 3.11 support

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

lpsinger reviewed (changes requested):
pytest-mpl 0.16.1 requires Python >= 3.6.

nilason commented:
> pytest-mpl 0.16.1 requires Python >= 3.6.

As pytest 7.4 requires Python >=3.7, how about to set the lowest Python version to 3.7?

nilason commented:
> pytest-mpl 0.16.1 requires Python >= 3.6.

As pytest 7.4 requires Python >=3.7, how about to set the lowest Python version to 3.7?

nilason commented:
> > pytest-mpl 0.16.1 requires Python >= 3.6.
> 
> As pytest 7.4 requires Python >=3.7, how about to set the lowest Python version to 3.7?

Updated PR, removed Python 2.7.
